### PR TITLE
support 'polygon' query limiter, which smartly limits fetches based on box intersections of the polygon(s) given

### PR DESCRIPTION
**Pull request by hobu**

### CODE REVIEW

**Comment by msmitherdc** [comment]
Is the clip a actual POLYGON or does it use the POLYGON to create a bbox? And are MULTIPOLYGONs supported?

**Comment by hobu** [comment]
The idea is that it works the same as `filters.crop` but the polygon is used to pre-filter the EPT requests instead of testing the points. Whatever is supported in the `polygon` option of `filters.crop` would be supported as far as POLYGON/MULTIPOLYGON/GeoJSON, etc.

**Comment by msmitherdc** [comment]
Do you need to put something in to limit it to POLY/MULTIPOLYGON?

**Comment by abellgithub** [comment]
I don't understand the advantage of this over just adding a crop filter.  We're not eliminating any point reading with this.

**Comment by hobu** [comment]
Connor and I discussing that when you put a `polygon` on the `readers.ept`, you always want the equivalent of `filters.crop` output – ie, the points cropped too. That means baking in a `filters.crop` into the `readers.ept` or something.

**Comment by msmitherdc** [comment]
does it prevent pulling that data over the wire is the important question?

**Comment by hobu** [comment]
> does it prevent pulling that data over the wire is the important question?

Yes it does, but then you also want the points cropped to that same polygon. Why did you bother otherwise?

**Comment by msmitherdc** [comment]
Does the spatial reference parameter apply to this?

File: io/EptReader.cpp [hunk 1]
```diff
@@ -215,12 +216,19 @@ void EptReader::initialize()
     // Set geometry from polygons.
     if (m_args->m_polys.size())
     {
-        m_polys.clear();
+        m_queryPolys.clear();
         for (Polygon& poly : m_args->m_polys)
         {
             // Throws if invalid.
             poly.valid();
```
Comment (abellgithub) [inline io/EptReader.cpp]: valid just returns a bool.  It doesn't throw.

https://github.com/PDAL/PDAL/blob/master/pdal/Geometry.cpp#L176

**Comment by abellgithub** [comment]
@hobu: This could still be better as far as eliminating processing checks, but I think it's OK for a first cut.  Please have a look when you're back to work.

**Pull request merged**